File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxarr = 0;
        int l = 0;
        int r = height.size() - 1;
        while(l < r)
        {
            maxarr = max(maxarr, min(height[r], height[l]) * (r-l));
            if(height[l] < height[r])
                l++;
            else
                r--;
        }
        return maxarr;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxAr = 0
        l = 0
        r = len(height) - 1
        while l < r:
            width = r - l
            maxAr = max(maxAr, width * min(height[l],height[r]))
            
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return maxAr
            